### ISSUE TITLE
Base path not working like the docs said

### ISSUE THREAD

**Issue by harrytran998** [withastro/astro#5513]
### What version of `astro` are you using?

latest

### Are you using an SSR adapter? If so, which one?

NodeJs 18 LTS + SSR nodejs standalone

### What package manager are you using?

yarn 3

### What operating system are you using?

Mac

### Describe the Bug

https://docs.astro.build/en/reference/configuration-reference/#base

The docs said it only affects production. But seem in development, this still have effect.

When I config `astro.config.js` like this

```js
import { defineConfig } from 'astro/config';
import svelte from '@astrojs/svelte';
import AstroNode from '@astrojs/node';

export default defineConfig({
  output: 'server',
  integrations: [svelte()],
  base: '/ok',
  server: {
    port: 4000,
    host: '0.0.0.0',
  },
  adapter: AstroNode({
    mode: 'standalone',
  }),
});

```
In dev mode, it will throw a warning like this

```bash
[serve]  404   /favicon.ico
[dev] Requests for items in your public folder must also include your base. /favicon.svg should be /ok/favicon.svg. Omitting the base will break in production.
```

And the screen shows like this

```
404: Not found
In your site you have your base path set to [/ok/](http://localhost:3000/ok/). Do you want to go there instead?
```

### Link to Minimal Reproducible Example

Stackblitz not work correctly with base path. So your team need to pull this source code 😆 https://github.com/harrytran998/rep-astro-base-path

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by harrytran998** [comment]
> I had this issue, and did this in the build script as a workaround, the problem is setting basepath doesn't affect where Astro puts the build assets:
> 
> So make your base path directory in dist/ then copy/move the files from dist to the correct place.
> 
> ```
>     "build": "astro build && mkdir ./ok && cp -R ./dist/* ./ok && mv ./ok ./dist",
> ```
> 
> Then have your hrefs and src's point at /ok/theAsset.png/js etc
> 
> I spoke to Nate on their discord about this he was enquiring on if this was intended behavior.

Thanks for your suggestion 🙏. But seem we need official support for everyone can use this 😁.
I will try this soon.

**Comment by wassfila** [comment]
this feels like a painful issue, every Astro user has a different adhoc solution, here's what I ended up doing, a custom config file
```javascript
if(process.env.MODE == "MIDDLEWARE"){
    config.base = ""
}
```
https://github.com/MicroWebStacks/astro-big-doc/blob/dd2ef27ff03fa649f801f66e7e96a5e7334513b6/config.js#L13

because if I try to import the config in astro.config.mjs, then I get a real dev server crash, mystery, closed it only because it's vite, not Astro, but issue still valid.
https://github.com/withastro/astro/issues/5517

**Comment by marcfielding1** [comment]
Yeah I think their needs be a cross-platform solution to how basePath is applied - but this open source so we can all find a solution and contribute rather than relying on the core team @wassfila my company are happy to work on a proposal for this in terms of functionality but the trouble we'd need to deal with the various adaptors I think.(vercel, upcoming AWS, netfly etc)

**Comment by wassfila** [comment]
That's true @marcfielding1 , especially for improvements that break compatibility. For base and many others, Astro inherites Vites problems, see https://vitejs.dev/config/shared-options.html#base
So maybe looking forward on how Vite is evolving can help. Astro is a Vite user but also contributor.

**Comment by marcfielding1** [comment]
basePath seems wonky at the moment but without being involved in internal astro dev conversation I can't criticise really, so yeah lets see what the team come back with - it definitely needs work, just in line with the goals I guess.

**Comment by harrytran998** [comment]
> LOL, seems not to work for me in my case 😆. Bcs we have a web with the old path `/prefix/ok/*` --> Astro project will serve `ok/*`. 

Currently, I use the scripts to replace the path in `entry.mjs` to work around but looks so messy 😆. I hope the Astro team can investigate some time to do this.

**Comment by matthewp** [comment]
Moving to the docs repo since this is a docs issue. Using the base in development is intended behavior.
